src/Microsoft.ML.Api/DataViewConstructionUtils.cs

```diff
@@ -946,7 +946,17 @@ public override ValueGetter<TDst> GetGetter<TDst>()
             throw Contracts.ExceptNotImpl("Type '{0}' is not yet supported.", typeT.FullName);
         }
 
-        internal override Delegate GetGetterDelegate() => Utils.MarshalInvoke(GetGetter<int>, MetadataType.RawType);
+        internal override Delegate GetGetterDelegate()
```

Comment:
>GetGetterDelegate [](start = 35, length = 17)

I think there is an advantage to using Utils.MarshalInvoke. Since the return value is being cast to a Delegate anyway, would it work if we just changed the return type of GetGetter to be Delegate instead of ValueGetter<T>?

Reply:
used  Tom's tip to get around this

---
In reply to: [239574626](https://github.com/dotnet/machinelearning/pull/1836#discussion_r239574626) [](ancestors = 239574626)